crates/bevy_ui/src/widget/image.rs

```diff
@@ -107,6 +108,10 @@ pub fn update_image_content_size_system(
                     // multiply the image size by the scale factor to get the physical size
                     size: size.as_vec2() * combined_scale_factor,
                 }));
+                if let Some(mut style) = style {
+                    let Vec2 { x, y } = size.as_vec2();
+                    style.aspect_ratio = Some(x / y);
```

Comment:
shouldn't this be the other way around? Using the `style.aspect_ratio` controlling image sizing? I mean, I had the idea that the `Style` is always a _controlling_ component, never controlled by other components.

Reply:
The way the web does it is that the intrinsic aspect ratio is separate to the aspect ratio style.

Reply:
same, not a fan of changing the `Style` component. Also, if the user set a specific aspect_ratio, we shouldn't overwrite it

Reply:
> The way the web does it is that the intrinsic aspect ratio is separate to the aspect ratio style.

I didn't find other place to set correct ratio, taffy knows nothing about type of nodes, the image should keep ratio, but div has no such restricktions. I agree that there shouldn't be overwriting, I will fix this.

Reply:
> I didn't find other place to set correct ratio, if you know such way, please, feel free to describe it, taffy knows nothing about type of nodes

Taffy needs to change to fix this sensibly. Either Taffy needs to allow "measurable" items to communicate their inherent aspect ratio. Or Taffy needs to pass the styles in to the measure function to allow measurable items to apply it themselves.

Reply:
So we will not fix examples until Taffy become ready?